### PR TITLE
Adding ability to use PFX files

### PR DESCRIPTION
**Pull request by tomgco**
Instead of extending ClientSSLSecurity to handle keys, certs, etc plus pfx keys, I have instead opted to create a new class specifically to use pfx files.

### CODE REVIEW

**Comment by jsdevel** [comment]
I like this strategy.  Can you add support for passwords?  I feel that may need to be an optional 3rd param.

**Comment by tomgco** [comment]
> I like this strategy. Can you add support for passwords? I feel that may need to be an optional 3rd param.

Sure thing

**Comment by tomgco** [comment]
Oops, failing for older versions of node, will fix

**Comment by tomgco** [comment]
Ok @jsdevel,  I have added in passphrase support for you, tests are also now passing.

File: lib/security/ClientSSLSecurityPFX.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+var fs = require('fs')
+  , https = require('https')
+  , _ = require('lodash');
+
+/**
+ * activates SSL for an already existing client
+ *
+ * @module ClientSSLSecurityPFX
+ * @param {Buffer|String}   key
+ * @param {Buffer|String}   cert
```
Comment (jsdevel) [inline lib/security/ClientSSLSecurityPFX.js]: can you update this jsdoc?

File: test/security/ClientSSLSecurityPFX.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+describe('ClientSSLSecurityPFX', function() {
+  var ClientSSLSecurityPFX = require('../../').ClientSSLSecurityPFX;
+  var pfx = __filename;
+
+  it('is a function', function() {
+    ClientSSLSecurityPFX.should.be.type('function');
+  });
+
+  describe('defaultOption param', function() {
+    it('is accepted as the second param', function() {
```
Comment (jsdevel) [inline test/security/ClientSSLSecurityPFX.js]: `it('should....` is conventional.

File: test/security/ClientSSLSecurityPFX.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      //should happen!
+      instanceCert = false;
+    }
+
+    if (instanceCert !== false) {
+      throw new Error('accepted wrong pfk');
+    }
+  });
+
+  it('should be able to make a request', function (done) {
```
Comment (jsdevel) [inline test/security/ClientSSLSecurityPFX.js]: Will `ClientSSLSecurityPFX` make a request, or will the library as a whole?  A more appropriate spec name may be `it('should be usable in a request...`.

**Comment by jsdevel** [comment]
Pretty close @tomgco !  Once the last minor comments are addressed we'll merge.

**Comment by tomgco** [comment]
> Pretty close @tomgco ! Once the last minor comments are addressed we'll merge.

Thanks, just adding these changes now.

**Comment by tomgco** [comment]
Ok, @jsdevel should be all working now and up to scratch.

Want me to rebase my commits into one, or is it good as they are?

**Comment by herom** [comment]
Thanks a lot for your time and effort @tomgco :+1: 

Please rebase and squash your commits into a single one as stated in our Guidelines on [Submitting a Pull Request](https://github.com/vpulim/node-soap/blob/master/CONTRIBUTING.md#submitting-a-pull-request) in order to get your PR merged :smiley:

**Comment by tomgco** [comment]
We are all rebased and should be ready to go, thanks @herom and @jsdevel!

**Comment by herom** [comment]
Thanks a ton @tomgco - it's people like you who make this library so awesome :smiley:

**Comment by jsdevel** [comment]
> Thanks a ton @tomgco - it's people like you who make this library so awesome :smiley:

+1

**Comment by herom** [comment]
@tomgco Actually tests are failing over at TravisCI because `client-password.pfx` has expired and therefore the test "ClientSSLSecurityPFX should be usable in a request" (see [here](https://github.com/vpulim/node-soap/blob/master/test/security/ClientSSLSecurityPFX.js#L43)) throws an error.

Since you introduced this valuable extension and provided the files for testing would it be possible for you to add a valid file so that the tests are passing? Thanks a lot :+1:

**Comment by jsdevel** [comment]
@tomgco can you re-generate the `pfx` files with a _really_ long expire time or upload the command you used to create them?

**Comment by jsdevel** [comment]
It would be nice if we scripted these with bash or something.

**Comment by tomgco** [comment]
@jsdevel Hi, sorry about this, I haven’t got the scripts on hand at the moment but should be able to upload a script and a new cert come the weekend.

**Comment by herom** [comment]
@tomgco :+1: 

@jsdevel should we restart all tests on currently open PRs and merge them since you've commented out the currently invalid test?

**Comment by jsdevel** [comment]
> should be able to upload a script and a new cert come the weekend.

Thanks @tomgco!  That would be great :smile: 

> should we restart all tests on currently open PRs and merge them since you've commented out the currently invalid test?

I think we'll need to call for a rebase first, otherwise the restart will continue to fail with the test uncommented.  But yes, we should notify other PR authors.

**Pull request merged**